ClassMoodApp/static/tsc/controllers/LectureListController.ts

```diff
@@ -0,0 +1,126 @@
+/// <reference path="../app.ts" />
+/// <reference path="../../../../typings/bootbox/bootbox.d.ts" />
+declare var angular: ng.IAngularStatic;
+module ClassMoodApp {
+    "use strict";
+    export class LectureListController {
+        static $inject = ["$scope", "$http", "$window", "$routeParams"];
+        public lectures:Array<LectureListModel>;
+        public classId: number;
+        public isStudent: boolean;
+        private userId: number;
+        constructor(private $scope: ng.IScope,
+                    private $http: ng.IHttpService,
+                    private $window: ng.IWindowService,
+                    private $routeParams: ng.route.IRouteParamsService ) {
+                        this.$http = $http;
+                        this.$scope = $scope;
+                        this.$window = $window;
+                        this.$routeParams = $routeParams;
+                        // console.log(this.$routeParams)
+                        // this.classId = this.$routeParams['classId']
+                        this.lectures = [];
+                        // this.getCurrentlectures();
+                        // this.getIsProfessor();
+                        // this.getUserId();
+                    }
+
+        public init(classId: number): void {
+            this.classId = classId;
+            this.getCurrentlectures();
+            this.getIsProfessor();
+            this.getUserId();
+        }
+
+        private getCurrentlectures(): void {
+            this.$http.get(`/get_lecture_list/${this.classId}`).success(
+                (data: any, status) => {
+                    this.lectures = data.results;
+                }
+            ).catch((error) => {
+                console.log(error);
+            });
+        }
+
+        private getUserId(): void {
+            this.$http.get('/user_id').then(
+                (response: any) => {
+                    this.userId = response.results;
+                }
+            )
+        }
+
+        private getIsProfessor(): void {
+            this.$http.get("/is_student").success(
+                (is_student: any, status) => {
+                    this.isStudent = is_student.results;
+                }
+            )
+        }
+
+        public addLecture(): void {
+            bootbox.prompt({
+                title: "Enter Lecture Name",
+                value: 'Lecture Name',
+                callback: (lectureName: string) => {
+                    if (lectureName != null) {
+                        if (this.isStudent !== true) {
+                            this.$http.get(`/add_lecture/${this.classId}/${lectureName}`).then(
+                                (response: any) => {
+                                    console.log(response.data);
+                                }
+                            )
+                        }
+                    } 
+                }
+            });
+            this.$window.location.reload();
```

Comment:
Do not use reload. Instead make an update function, as in this case, every time u call reload, the entire page has to reload. With an update function, only the table elements will change which will be quicker.